pkgs/applications/networking/irc/konversation/1.6.nix

```diff
@@ -0,0 +1,65 @@
+{ stdenv, lib, fetchurl, cmake, kde5, makeQtWrapper, phonon}:
+
+let
+  pn = "konversation";
+  v = "1.6";
+
+  buildInputs = with kde5; [
+    cmake
+    extra-cmake-modules
+    kbookmarks
+    karchive
+    kconfig
+    kconfigwidgets
+    kcoreaddons
+    kdbusaddons
+    kdoctools
+    kemoticons
+    kglobalaccel
+    ki18n
+    kiconthemes
+    kidletime
+    kitemviews
+    knotifications
+    knotifyconfig
+    kio
+    kparts
+    kwallet
+    solid
+    sonnet
+    phonon
+  ];
+in
+
+stdenv.mkDerivation rec {
+  name = "${pn}-${v}";
+
+  src = fetchurl {
+    url = "mirror://kde/stable/${pn}/${v}/src/${name}.tar.xz";
+    sha256 = "789fd75644bf54606778971310433dbe2bc01ac0917b34bc4e8cac88e204d5b6";
+  };
+
+  nativeBuildInputs = [
+    kde5.extra-cmake-modules
+    kde5.kdoctools
+    makeQtWrapper
+  ];
+
+  inherit buildInputs;
+
+  cmakeFlags =
+    (buildInputs.cmakeFlags or [])
```

Comment:
ah yes. When I began writing this expression I thought I might want to base it on existing ones, so I copied this from `kde-app.nix`. I was wondering how this works but I think I know it now.

`kde-app.nix` is a function, that returns a function. The latter function is obtained in `kde-apps-15.12/default.nix` after which it is used in the KDE applications. `args` is then, if I am correct the whole set of arguments. In certain cases, one might want to give custom `cmakeFlags`, and in that case, that list with flags is used, and else an empty list.